### PR TITLE
Components

### PR DESCRIPTION
**Pull request by MarinaStepanchuk**
Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module01
Screenshot: 
Screenshot_248
Screenshot_247
Screenshot_246
Screenshot_249

Deploy: [movie-search-rs.netlify.app](https://movie-search-rs.netlify.app/)
Done 20.03.2023 / deadline 20.03.2023
Score: 15 / 15

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (Ksarelto) [inline src/App/App.tsx:1]: no need in this import

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { Pages, Paths } from '../../constants/common.constants';
+import React from 'react';
```
Comment (Ksarelto) [inline src/components/Header/Header.tsx:2]: no neeeeed, remo React import rom all components))

File: src/components/SearchBar/SearchBar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import SearchBar from './SearchBar';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { vitest } from 'vitest';
+import userEvent from '@testing-library/user-event';
+
+describe('SearchBar', () => {
+  const onTextChange = vitest.fn();
+
+  it('should render correct', async () => {
```
Comment (Ksarelto) [inline src/components/SearchBar/SearchBar.test.tsx:10]: if u describe test as render correct so show a snapshot or check that all components are rendered, but check that something was called is another story

File: src/components/SearchBar/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Buttons } from '../../constants/common.constants';
+import classes from './SearchBar.module.scss';
+
+interface ISearchBarProps {
+  text: string;
+  onTextChange: (value: string) => void;
+}
+
+class SearchBar extends React.Component<ISearchBarProps, object> {
```
Comment (Ksarelto) [inline src/components/SearchBar/SearchBar.tsx:10]: do u need the second argument in generic type?

File: src/components/SearchBar/SearchBar.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(public props: ISearchBarProps) {
+    super(props);
+    this.handleTextChange = this.handleTextChange.bind(this);
```
Comment (Ksarelto) [inline src/components/SearchBar/SearchBar.tsx:13]: use arrow functions to avoid binding

File: src/components/SearchBar/SearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  private handleTextChange(e: React.ChangeEvent<HTMLInputElement>): void {
+    this.props.onTextChange(e.target.value);
+  }
+
+  public render(): JSX.Element {
+    const text = this.props.text;
```
Comment (Ksarelto) [inline src/components/SearchBar/SearchBar.tsx:21]: for what?) u can use it directry in jsx

File: src/components/ItemMovie/ItemMovie.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import IMovie from '../../interfaces/interfaces';
+import classes from './ItemMovie.module.scss';
+
+interface IItemMoviesProps {
+  key: number;
+  movie: IMovie;
+  text: string;
+}
+
+class ItemMovie extends React.Component<IItemMoviesProps, object> {
```
Comment (Ksarelto) [inline src/components/ItemMovie/ItemMovie.tsx:12]: the second argument in generic?

File: src/components/ItemMovie/ItemMovie.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  public render(): JSX.Element | undefined {
+    const { name, poster, year, countries, rating } = this.props.movie;
+    const text = this.props.text;
```
Comment (Ksarelto) [inline src/components/ItemMovie/ItemMovie.tsx:15]: use directly

File: tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "lib": ["dom", "esnext", "dom.iterable"],
+    "jsx": "react",
```
Comment (Ksarelto) [inline tsconfig.json:5]: u can set here react-jsx and not use imports in components

**Review by Ksarelto — COMMENTED** [review]
Very cool! Good job!

File: src/components/SearchBar/SearchBar.test.tsx (hunk 3 above)
Comment (MarinaStepanchuk) [reply src/components/SearchBar/SearchBar.test.tsx:10]: corrected describe )

File: src/components/SearchBar/SearchBar.tsx (hunk 4 above)
Comment (MarinaStepanchuk) [reply src/components/SearchBar/SearchBar.tsx:10]: no, removed)

File: src/components/SearchBar/SearchBar.tsx (hunk 5 above)
Comment (MarinaStepanchuk) [reply src/components/SearchBar/SearchBar.tsx:13]: I after all can write down bind directly at transfer of function in onChange? corrected like this.

File: src/components/SearchBar/SearchBar.tsx (hunk 6 above)
Comment (MarinaStepanchuk) [reply src/components/SearchBar/SearchBar.tsx:21]: removed)

File: src/App/App.tsx (hunk 1 above)
Comment (MarinaStepanchuk) [reply src/App/App.tsx:1]: I am using class components. I need it)

File: src/components/Header/Header.tsx (hunk 2 above)
Comment (MarinaStepanchuk) [reply src/components/Header/Header.tsx:2]: I am using class components. I need it)

File: src/components/ItemMovie/ItemMovie.tsx (hunk 8 above)
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx:15]: this value is used in several places.

File: tsconfig.json (hunk 9 above)
Comment (MarinaStepanchuk) [reply tsconfig.json:5]: corrected

**Pull request merged**